src/math/trig.c

```diff
@@ -58,16 +53,16 @@ void cordic_approx(int32_t th_rad_fxp, int32_t a_idx, int32_t *sign, int32_t *b_
 	 * without overflow.Increase of fractionLength makes the addition or
 	 * subtraction of a multiple of pi/2 more precise
 	 */
-	if (th_rad_fxp > cord_sincos_piovertwo_q28fl) {
-		if ((th_rad_fxp - cord_sincos_piovertwo_q29fl) <= cord_sincos_piovertwo_q28fl) {
-			th_rad_fxp -= cord_sincos_piovertwo_q29fl;
+	if (th_rad_fxp > PIOVERTWO_Q4_28) {
+		if ((th_rad_fxp - PIOVERTWO_Q3_29) <= PIOVERTWO_Q4_28) {
```

Comment:
I have an idea which might reduce some calculations: replace the statement
        (th_rad_fxp - PIOVERTWO_Q3_29) <= PIOVERTWO_Q4_28
by
        th_rad_fxp <= THREEQUARTERSOFPI_Q3_29
where THREEQUARTERSOFPI_Q3_29 = (PIOVERTWO_Q3_29 + PIOVERTWO_Q4_28) and also a constant